mesa_purchase_autopopulate/models/purchase/order.py

```diff
@@ -0,0 +1,126 @@
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    analytic_account_id = fields.Many2one(
+        "account.analytic.account",
+        compute="_compute_analytic_account",
+        inverse="_inverse_analytic_account",
+        string="Analytic Account",
+        store=True,
+    )
+    project_id = fields.Many2one(
+        "project.project", string="Project", related="sale_order_id.project_id"
+    )
+    sale_order_id = fields.Many2one("sale.order", string="Sale Order")
+    user_id = fields.Many2one(
+        compute="_compute_user_id", inverse="_inverse_user_id", store=True
+    )
+    force_analytic_account_id = fields.Many2one(
+        "account.analytic.account", string="Analytic Account (Custom)", store=True
+    )
+    force_user_id = fields.Many2one(
+        "res.users", string="Responsible Salesperson (Custom)", store=True
+    )
+
+    @api.multi
+    def action_view_invoice(self):
+        """
+        Override super to set the Vendor Bill's "Purchase Representative" to
+        match the related Sale Order's "Purchase Representative".
+
+        :return {dict}:
+            A dictionary mapping of fields and their values for the
+            Window Action of the record to be displayed or created
+        """
+        action = self.env.ref("account.action_vendor_bill_template")
+        result = action.read()[0]
+        create_bill = self.env.context.get("create_bill", False)
+        # override the context to get rid of the default filtering
+        result["context"] = {
+            "type": "in_invoice",
+            "default_purchase_id": self.id,
+            "default_currency_id": self.currency_id.id,
+            "default_company_id": self.company_id.id,
+            "company_id": self.company_id.id,
+        }
+        # choose the view_mode accordingly
+        if len(self.invoice_ids) > 1 and not create_bill:
+            result["domain"] = "[('id', 'in', " + str(self.invoice_ids.ids) + ")]"
+        else:
+            res = self.env.ref("account.invoice_supplier_form", False)
+            form_view = [(res and res.id or False, "form")]
+            if "views" in result:
+                result["views"] = form_view + [
+                    (state, view) for state, view in action["views"] if view != "form"
+                ]
+            else:
+                result["views"] = form_view
+                ### BEGIN CUSTOM CODE ###
+            if create_bill:
+                # If we want to create a Vendor Bill, Set the Purchase Order's
+                # Purchase Representative as the Vendor Bill's default
+                # Purchase Representative.
+                result["context"]["default_user_id"] = self.user_id.id
+                ### END CUSTOM CODE ###
+            else:
+                # Do not set an invoice_id if we want to create a new bill.
+                result["res_id"] = self.invoice_ids.id or False
```

Comment:
I think this will help others see what you've done a bit better.  It took me a minute to compare it to Core to see what you did.  :slightly_smiling_face: 
```suggestion

            ### BEGIN CUSTOM CODE ###
            if create_bill:
                # If we want to create a Vendor Bill, Set the Purchase Order's
                # Purchase Representative as the Vendor Bill's default
                # Purchase Representative.
                result["context"]["default_user_id"] = self.user_id.id
            else:
                # Do not set an invoice_id if we want to create a new bill.
                # if not create_bill:
                ### END CUSTOM CODE ###
                result["res_id"] = self.invoice_ids.id or False

```

Reply:
> Which PR were those notes from?

The Purchase Autopopulate Module.